test/src/test/java/jenkins/security/AgentSecretActionTest.java

```diff
@@ -0,0 +1,127 @@
+package jenkins.security;
+
+
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import hudson.model.Computer;
+import hudson.model.User;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.SlaveComputer;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+import org.kohsuke.stapler.StaplerRequest2;
+import org.kohsuke.stapler.StaplerResponse2;
+
+public class AgentSecretActionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private AgentSecretAction action;
+    private StaplerRequest2 req;
+    private StaplerResponse2 rsp;
+    private StringWriter stringWriter;
+    private PrintWriter writer;
+
+    @Before
+    public void setUp() throws Exception {
+        SlaveComputer computer = mock(SlaveComputer.class);
+        action = new AgentSecretAction(computer);
+        req = mock(StaplerRequest2.class);
+        rsp = mock(StaplerResponse2.class);
+        stringWriter = new StringWriter();
+        writer = new PrintWriter(stringWriter);
+        when(rsp.getWriter()).thenReturn(writer);
+    }
+
+    @Test
+    public void testGetSecretWithValidPermissions() throws Exception {
+        DumbSlave agent = j.createSlave("test-agent", null);
```

Comment:
I don't think that this will create an agent with a JNLPLauncher. Looking at the code it creates an agent with a specific launcher.